jerry-core/ecma/builtin-objects/ecma-builtin-array.inc.h

```diff
@@ -19,6 +19,8 @@
 
 #include "ecma-builtin-helpers-macro-defines.inc.h"
 
+#ifndef CONFIG_DISABLE_ARRAY_BUILTIN
+
```

Comment:
The generator checks all `.c` and `.h` files in the project, `.inc.h` included.

And in theory, yes, we could detect such things by replicating a lot of the behaviour of a standard preprocessor. In practice, however, I don't intend to write a new preprocessor. The generator does not deal with (analyse, track) interfile dependencies (who includes whom, what are system includes, what are project includes, what are the include paths, etc. -- this would be just way too much). As mentioned above:

> writing a context-sensitive (include-tracking) analyser -- but writing something like that just for this task would be an overkill IMHO.

Actually, I did take a look at the preprocessor of gcc whether it could output something useful for our purpose (which lines are guarded by which preprocessor conditions) -- as preprocessors do/can output quite some info in addition to the preprocessed source --, but it turned out that nothing similar is available. (Most extra dumps are for helping makefile dependency tracking.)